docs/ref/models/indexes.txt

```diff
@@ -174,3 +174,59 @@ See the PostgreSQL documentation for more details about `covering indexes`_.
     <django.contrib.postgres.indexes.GistIndex>`.
 
 .. _covering indexes: https://www.postgresql.org/docs/current/indexes-index-only-scans.html
+
+
+The ``*expressions`` argument is a list of positional expressions that the
+index will be based on.
+``Index(Lower('title').desc(), 'pub_date', name='lower_title_date_idx')``
+would create an index on the lowercase value of the ``title`` field in
+descending order and the ``pub_date`` field in the default ascending order.
+When using expressions in combination with ordering,
+:class:`~django.db.models.functions.Collate` and
+:class:`~django.contrib.postgres.indexes.OpClass` it's important that they are
+the topmost expressions, i.e ``Index(Lower(F('name').desc()), name='name_idx')``
+is not allowed. Instead it should be written as
+``Index(Lower(F('name')).desc()``.
+Using ``F`` expressions in a construct like
+``Index(F('height') * F('weight'), name='math_idx')`` allows
+constructing indexes on mathematical operations.
+
+.. admonition:: Database specific exceptions of suitable expressions
+
+     :PostgreSQL:
+         PostgreSQL requires that "*all functions and operators used in an index
+         definition must be 'immutable'*" --- `PostgreSQL documentation
+         <https://www.postgresql.org/docs/current/sql-createindex.html#AEN80057>`_.
```

Comment:
I can't find the `#AEN80057` anchor in the targeted page. We can remove it or use the anchor for the description section `#id-1.9.3.69.5`

Reply:
Super!